packages/apptive_grid_heinzelmen/lib/src/data_widget/data_widget.dart

```diff
@@ -36,284 +36,219 @@ class DataWidget extends StatelessWidget {
-      );
-    }
-
-    if (data is PhoneNumberDataEntity) {
-      return InkWell(
-        onTap: () {
-          (linkLauncher ?? const LinkLauncher()).openWebPage(
-            url: Uri.parse('tel:${data.value}'),
-            openExternally: true,
-          );
-        },
-        child: Text(
-          data.value.toString(),
-          style: (textStyle ?? const TextStyle()).copyWith(
-            color: Theme.of(context).primaryColor,
-            decoration: TextDecoration.underline,
+      CurrencyDataEntity() => Text(
+          NumberFormat.currency(
+            locale: locale,
+            symbol: NumberFormat()
+                .simpleCurrencySymbol((data as CurrencyDataEntity).currency),
+          ).format(data.value),
+          style: textStyle,
+        ),
+      UriDataEntity() => InkWell(
+          onTap: () {
+            (linkLauncher ?? const LinkLauncher())
+                .openWebPage(url: data.value, openExternally: true);
+          },
+          child: Text(
+            data.value.toString(),
+            style: (textStyle ?? const TextStyle()).copyWith(
+              color: Theme.of(context).primaryColor,
+              decoration: TextDecoration.underline,
+            ),
           ),
         ),
-      );
-    }
-
-    if (data is EmailDataEntity) {
-      return InkWell(
-        onTap: () {
-          (linkLauncher ?? const LinkLauncher()).openWebPage(
-            url: Uri.parse('mailto:${data.value}'),
-            openExternally: true,
-          );
-        },
-        child: Text(
-          data.value.toString(),
-          style: (textStyle ?? const TextStyle()).copyWith(
-            color: Theme.of(context).primaryColor,
-            decoration: TextDecoration.underline,
+      EmailDataEntity() => InkWell(
+          onTap: () {
+            (linkLauncher ?? const LinkLauncher()).openWebPage(
+              url: Uri.parse('mailto:${data.value}'),
+              openExternally: true,
+            );
+          },
+          child: Text(
+            data.value.toString(),
+            style: (textStyle ?? const TextStyle()).copyWith(
+              color: Theme.of(context).primaryColor,
+              decoration: TextDecoration.underline,
+            ),
           ),
         ),
-      );
-    }
+      PhoneNumberDataEntity() => InkWell(
+          onTap: () {
+            (linkLauncher ?? const LinkLauncher()).openWebPage(
+              url: Uri.parse('tel:${data.value}'),
+              openExternally: true,
+            );
+          },
+          child: Text(
+            data.value.toString(),
+            style: (textStyle ?? const TextStyle()).copyWith(
+              color: Theme.of(context).primaryColor,
+              decoration: TextDecoration.underline,
+            ),
+          ),
+        ),
+      SignatureDataEntity() => AspectRatio(
+          aspectRatio: 2,
+          child: Thumbnail(attachment: data.value),
+        ),
+      DataEntity<DataEntity, dynamic>() => DataWidget(
```

Comment:
The fact that the sealed class recognises this for all the cases where the value is actually a DataEntity is just so nice